more contribution guide updates

stumblefiend opened the pull request:
<!-- readthedocs-preview writethedocs-www start -->
----
📚 Documentation preview 📚: https://writethedocs-www--2173.org.readthedocs.build/

<!-- readthedocs-preview writethedocs-www end -->

stumblefiend committed:
more contribution guide updates

stumblefiend committed:
create section on simple edits through GitHub UI

stumblefiend committed:
fix bullet list indentation

stumblefiend committed:
update another bullet list

plaindocs commented on docs/guide/contributing.rst:
```diff
@@ -31,37 +31,35 @@ We especially need contributions in these areas:
+-------------------------
 
-Follow these guidelines for discussing tools: 
+General guidelines include:
```
```suggestion

```

Somewhat tongue in cheek, but also not really

plaindocs reviewed (approved):
Seems good.

CollierCZ commented on docs/guide/contributing.rst:
```diff
@@ -2,27 +2,27 @@
-Where to start
---------------
+Where to contribute
+--------------------
```
```suggestion
What to contribute
-------------------
```

stumblefiend replied:
Thanks for the recommendation. I was definitely considering to update and will do so since you are obviously in agreement.

CollierCZ commented on docs/guide/contributing.rst:
```diff
@@ -31,37 +31,35 @@ We especially need contributions in these areas:
-* Avoid advocacy and plugs about your favorite toolchain, even if it's open source.
-* Present specific use cases, how problems were solved, and what worked or didn't work well. 
-* Consider to first `file a GitHub issue`_ or contact one of the guide editors at guide@writethedocs.org.
+* Use a friendly and encouraging tone. It's a good way to write docs, so practice with a guide at Write the Docs!
```
```suggestion
* Use a friendly and encouraging tone. It's a good way to write docs, so practice with docs about docs!
```
I think the original was a nice turn of phrase. Not a requirement or anything.

stumblefiend replied:
That's strange. I honestly have no idea where that "docs about docs" thing came from. I don't recall writing it. It may have been a copy/paste from somewhere else in the doc because there was a lot of repeat content. I'll definitely investigate further.

CollierCZ replied:
Seems like it's [been there for 6 years](https://github.com/writethedocs/www/blame/32ef981f5490c5dc99c68afc15203eea3712d100/docs/guide/contributing.rst#L44). 🤷

stumblefiend replied:
Oh, thanks for taking a look. I had it backwards...I thought you were saying I had added the "docs about docs" line. All good now!

CollierCZ commented on docs/guide/contributing.rst:
```diff
@@ -31,37 +31,35 @@ We especially need contributions in these areas:
-Prerequisites
---------------
+Prerequisites for contributing
+-------------------------------
```
I'm not sure I understand the need to make this longer. Could you explain? `Prerequisites` (or more informally `Before you start`) is a common heading in how-to guides, I think, and is usually clear enough on it's own, I would hope.

CollierCZ replied:
Especially true give that these aren't prerequisites for contributing, but for contributing in a certain way.

stumblefiend replied:
Since, as you mentioned, these are prerequisites for contributing a certain way, should we actually clarify even further then? Something like "Prerequisites for contributing via git"? Otherwise, as you mentioned in another comment that the structure of the content should make the context of the prerequisites obvious, we should just move the prerequisites section under the "Update a guide in Write the Docs via a pull request" section.

stumblefiend replied:
Also, we should be cognizant that many individuals will come to know about a section of a doc via a search on the website and that we should err towards extra context most of the time. Plus, SEO will care about how we name our headers such that we may need to be more redundant despite the context of the article (not that either of these arguments matter for this particular situation).  Just letting the group know where I'm coming from when thinking about header text. 

![Screenshot 2024-07-11 8 18 58 AM](https://github.com/writethedocs/www/assets/64424206/4dd995ea-366e-4b50-9900-ef996a8af303)

CollierCZ replied:
I don't think that someone will be searching for "prerequisites" but itself. I feel like SEO might actually be an argument for not having additional context in there. Because people shouldn't be coming to that heading on its own, just within the context.

Take an example from [Google's cloud docs[(https://cloud.google.com/run/docs/quickstarts/build-and-deploy/deploy-python-service) (which I assume would have some insight into the SEO advantages of headings). They consistently use `Before you begin`.

It's not something I feel would be blocking either way, just saying. 🙂

stumblefiend replied:
I know they won't search for "prerequisites" by itself. I was providing an example to show one of the many reasons why we need to be careful about headers in general. 

The biggest takeaway from the example you provided from Google is that we shouldn't be using formal, complicated language like "prerequisites" at all and should use simpler text like "before you begin." Do you agree? Thanks for sharing the example!

CollierCZ replied:
Yeah, I like it better. I remembered it as `Before you start` in my first message in this thread, but really either way is fine.

CollierCZ commented on docs/guide/contributing.rst:
```diff
@@ -71,11 +69,28 @@ Prerequisites
+Editing a single guide directly in the GitHub UI
+--------------------------------------------------
+
+1. :ref:`Complete the prerequisites <prerequisites>`.
```
I see a couple problems with this:
* These aren't requirements for using the UI (only for the terminal)
* I don't think the first step in a guide should ever be "complete the prerequisites". I think that should be clear enough from the general structure.
Only the first is really a blocking issue.

CollierCZ commented on docs/guide/contributing.rst:
```diff
@@ -71,11 +69,28 @@ Prerequisites
+
+1. :ref:`Complete the prerequisites <prerequisites>`.
+
+2. Click the pencil icon to edit this file in your fork of the www repo.
```
```suggestion
2. Click the pencil icon to create a fork of the `www` repo and start editing the file in it.
```
Assuming it's the first edit (since otherwise, they probably don't need the guide).

CollierCZ commented on docs/guide/contributing.rst:
```diff
@@ -86,13 +101,15 @@ If you've never worked with Git and GitHub, use this section to get started.
 **In the GitHub UI:**
 
-1.  `Fork the Write the Docs www project`_.
+1. :ref:`Complete the prerequisites <prerequisites>`.
```
As above, I don't think this step adds clarity.

CollierCZ reviewed (changes requested):
I like the initiative to make things clearer. 👍 

I'm not sure I would give preference to the UI version, as the suggestion to learn the tools is a good one and it really is more difficult for complex changes, but that's just a personal preference and not blocking or anything.

But the addition of the UI version makes the flow not work anymore, since the prerequisites aren't for the UI version. So the overall flow of the guide needs adjusting to that fact.

stumblefiend committed:
Update docs/guide/contributing.rst

Co-authored-by: Aaron Collier <aaron@collier.cz>

stumblefiend committed:
Update docs/guide/contributing.rst

Co-authored-by: Sam <lykoszine@gmail.com>

stumblefiend commented:
> I like the initiative to make things clearer. 👍
> 
> I'm not sure I would give preference to the UI version, as the suggestion to learn the tools is a good one and it really is more difficult for complex changes, but that's just a personal preference and not blocking or anything.
> 
> But the addition of the UI version makes the flow not work anymore, since the prerequisites aren't for the UI version. So the overall flow of the guide needs adjusting to that fact.

If that's the case, can I also remove that line about "if you don't want to work with git then email us some content to update."  In this day and age, we need both our technical and non-technical documentarians to step up their skills if they haven't done so by now. Understanding Git is crucial in most cases or, at worst, a highly-desirable skill. The tough love would be for their own good. Thoughts?

UPDATE --> I actually went ahead and made a commit that addresses a lot of the comments from the review and also removes the statement about them emailing us to provide recommended changes outside of git.

stumblefiend committed:
Update docs/guide/contributing.rst

Co-authored-by: Aaron Collier <aaron@collier.cz>

stumblefiend committed:
Update docs/guide/contributing.rst

Co-authored-by: Aaron Collier <aaron@collier.cz>

stumblefiend committed:
updates based on suggestions from reviewers

CollierCZ commented:
To be clear, I didn't mean to push for not including the UI at all as an option. I use it myself and it can be a way to ease into Git. I just wanted to make sure the page worked as a whole. I think it'd be fine to put the steps you had before about the UI back in.

The only thing I'd say in that regard is that these two options:

- Edit a single guide directly in GitHub.
- For changes that span multiple files [create a fork, a branch, and a pull request](https://writethedocs-www--2173.org.readthedocs.build/guide/contributing/#update-guide-in-git).

aren't mutually exclusive. Even if you edit a single guide directly in GitHub, you have to create a fork, a branch, and a pull request. That's why I think of it as easing into learning Git, rather than avoiding it with email or whatever.

So I think it might be good to have the option for the UI, but up to you. Just wouldn't present it as not needing a PR.

CollierCZ commented on docs/guide/contributing.rst:
```diff
@@ -2,27 +2,24 @@
+There are several ways to contribute to the content:
 
+* Edit a single guide directly in GitHub.
 * For changes that span multiple files :ref:`create a fork, a branch, and a pull request <update-guide-in-git>`.
```
```suggestion
* For changes that span multiple files, :ref:`create a fork, a branch, and a pull request <update-guide-in-git>`.
```
Not your change, but while we're in here. 😉

CollierCZ commented on docs/guide/contributing.rst:
```diff
@@ -192,7 +187,7 @@ The Write the Docs community is available for help, questions, or discussion:
 .. _file a GitHub issue: https://github.com/writethedocs/www/issues/new
 .. _Submit an issue: https://github.com/writethedocs/www/issues/new
-.. _reStructuredText (.rst) Markup: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html
+.. _reStructuredText (.rst): https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html
```
Just noticed, is there a reason the Markdown link goes to the WTD intro but the rST one doesn't? There is a [WTD intro to rST](https://writethedocs-www--2173.org.readthedocs.build/guide/writing/reStructuredText/). Not a big deal, just seemed curious. 🤷

stumblefiend replied:
I'm not sure. I thought I used whatever links were already present in the document. I'll review to see which feels like a better source to link to and adjust accordingly in my next commit. Thanks for bringing this up.

stumblefiend replied:
Updated in latest commit.

stumblefiend commented:
> To be clear, I didn't mean to push for not including the UI at all as an option. I use it myself and it can be a way to ease into Git. I just wanted to make sure the page worked as a whole. I think it'd be fine to put the steps you had before about the UI back in.
> 
> The only thing I'd say in that regard is that these two options:
> 
> * Edit a single guide directly in GitHub.
> * For changes that span multiple files [create a fork, a branch, and a pull request](https://writethedocs-www--2173.org.readthedocs.build/guide/contributing/#update-guide-in-git).
> 
> aren't mutually exclusive. Even if you edit a single guide directly in GitHub, you have to create a fork, a branch, and a pull request. That's why I think of it as easing into learning Git, rather than avoiding it with email or whatever.
> 
> So I think it might be good to have the option for the UI, but up to you. Just wouldn't present it as not needing a PR.

Sorry for the misunderstanding on that! I thought you were advocating not to include them. I will create a new commit that adds them back but still clarifies where the "prerequisites" apply.

stumblefiend committed:
Update docs/guide/contributing.rst

Co-authored-by: Aaron Collier <aaron@collier.cz>

CollierCZ commented:
> Sorry for the misunderstanding on that!

My fault for being unclear. I was mostly wondering if it should be first or second, but that doesn't matter too much in the broad scheme of things. As long as you add in the clarity about what's needed when. 🙂

stumblefiend commented:
> To be clear, I didn't mean to push for not including the UI at all as an option. I use it myself and it can be a way to ease into Git. I just wanted to make sure the page worked as a whole. I think it'd be fine to put the steps you had before about the UI back in.
> 
> The only thing I'd say in that regard is that these two options:
> 
> * Edit a single guide directly in GitHub.
> * For changes that span multiple files [create a fork, a branch, and a pull request](https://writethedocs-www--2173.org.readthedocs.build/guide/contributing/#update-guide-in-git).
> 
> aren't mutually exclusive. Even if you edit a single guide directly in GitHub, you have to create a fork, a branch, and a pull request. That's why I think of it as easing into learning Git, rather than avoiding it with email or whatever.
> 
> So I think it might be good to have the option for the UI, but up to you. Just wouldn't present it as not needing a PR.

Thanks for the clarity on your previous statements. I see the problem now and will update.

stumblefiend committed:
further updates based on review feedback

re-added UI instructions and made additional updates to some sentences for clarity and to shorten sentence length.

stumblefiend commented:
Fun fact, between the previous PR and this one, we've reduced the words from 1256 to 750 (that's even including the new section I added about the UI)!  Overall, our sentences are much shorter and clearer too. I think this is an overall win for readability, which is especially important for instructions that are key for folks to follow.

Now...if only we could encourage more contributions from our members, especially in terms of new content 😭 I dunno...maybe let's hold a contest or something for people to write guides for our site and we judge and award the best contributions (could be either a monetary reward or way they can get resume clout)? There is so much that needs updating in this portion of the site and I'm barely scratching the surface.

Notes for future improvements to the contribution guidelines: 

- We could say what the different formats for creating links are in rST and when to use each (even I'm not clear yet).
- We could mention how to look at the build details and what to look for.

CollierCZ commented on docs/guide/contributing.rst:
```diff
@@ -31,37 +27,52 @@ We especially need contributions in these areas:
+Editing a single guide via the GitHub UI
+-------------------------------------------
+
+1. Click the pencil icon to edit this file in your fork of the www repo.
```
Looking at this again, steps 4 & 5 from what used to be called the prerequisites are kind of necessary here. Sorry I was so focused on not forcing people to install Git when using only the UI. 🙈 
```suggestion
1. In the [Write the Docs www project](https://github.com/writethedocs/www), find the file name corresponding to an [existing issue](https://github.com/writethedocs/www/issues) or page you want to improve.

   For example, /docs/documentarians.rst creates https://www.writethedocs.org/documentarians/.

1. Review formatting guidelines for the Markup style the file uses. The example file above uses [reStructuredText (.rst)](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html).
1. Click the pencil icon to create a fork of the `www` repo and edit the file there.
```

Not a blocker, but might be nice 🤷

stumblefiend replied:
I included similar changes in my latest commit.

CollierCZ commented on docs/guide/contributing.rst:
```diff
@@ -2,27 +2,23 @@
-* For changes that span multiple files :ref:`create a fork, a branch, and a pull request <update-guide-in-git>`.
-* `Submit an issue`_ to suggest small changes, new content, or to file bugs against existing content.
+* :ref:`Edit a single guide directly in the GitHub UI <update-guide-via-github>`.
+* :ref:`Edit files via git <update-guide-in-git>`.
```
Minor nit
```suggestion
* :ref:`Edit files using Git <update-guide-in-git>`.
```

CollierCZ reviewed (approved):
Shorter and clearer is always a good combination!

There's a community board meeting coming up. I'll bring up the question there and see if there are any ideas or thoughts on the best way to encourage contributions. Definitely would be nice.

A couple of small things (one even smaller than small), but still looks good overall.

stumblefiend committed:
Update docs/guide/contributing.rst

Co-authored-by: Aaron Collier <aaron@collier.cz>

stumblefiend committed:
round 3 of updates from review feedback

stumblefiend committed:
fix styling for instructions
